cli/src/lib.rs

```diff
@@ -179,89 +35,32 @@ fn process_to_kml(
 }
 
 fn read_input_file(
-    cmd: &Command,
     args: &ArgMatches,
-    formats: &HashMap<&str, Format>,
     input_path: &Path,
 ) -> Result<track2kml::Database, anyhow::Error> {
-    let hint: Option<&String> = args.get_one("hint");
-
-    let formats = if let Some(hint) = hint {
-        if let Some(format) = formats.get(hint.as_str()) {
-            vec![format]
-        } else {
-            // We set possible values on the arg value parser, we should not be able to reach here
-            unreachable!()
-        }
-    } else {
-        // Check extensions
-        formats
-            .values()
-            .filter(|Format { file_ext, .. }| Some(file_ext) == input_path.extension().as_ref())
-            .collect()
-    };
-
-    if formats.is_empty() {
+    if input_path.extension().as_ref() != Some(&OsStr::new("json")) {
         return Err(anyhow::anyhow!(
             "Could not load input file.\n\
-    Cannot determine format from extension; try using `--hint`"
+    Invalid file extension; the file should be json."
```

Comment:
This should be a warning at most. Nothing prevents the program from continuing even if the extension is not JSON; plus we haven't really decided the format extension yet (we're using JSON right now but could be using anything else)